### ISSUE TITLE
Running Pode as Azure function, error 400 when posting information from html form

### ISSUE THREAD

**Issue by andrewrdavidson** [Badgerati/Pode#709]
### Question
I'm trying to run Pode server as an Azure Function, and all works as I expected until I tried to post form data to the Azure function. No matter what I seem to try I get 400 Bad Request back from Pode within the Azure Function.

Here is the form I'm using:

```
<html>
<head>
    <title>Form Test</title>
</head>
<body>
    <form action="http://localhost:7071/api/s" method="post" enctype="multipart/form-data">
        <div>
            <label>Username:</label>
            <input type="text" name="username" />
        </div>
        <div>
            <input type="submit" value="Submit" />
        </div>
    </form>
</body>
</html>
```

and the run.ps1 for the Azure Function:
```
using namespace System.Net
param($Request, $TriggerMetadata)
Start-PodeServer -Request $TriggerMetadata -ServerlessType AzureFunctions {
    Add-PodeRoute -Method POST -Path '/api/s' -ScriptBlock {
        Write-PodeJsonResponse -Value @{ "message" = "OK" }
    }
}
```

when starting I get the following verbose output:

```
func host start --verbose

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %


Azure Functions Core Tools
Core Tools Version:       3.0.3331 Commit hash: 0b57b1647c5a385beb8f7d64fa0e534ad4c7555d
Function Runtime Version: 3.0.15371.0

[2021-03-13T12:41:10.541Z] Building host: startup suppressed: 'False', configuration suppressed: 'False', startup operation id: '16584ae7-9853-4a63-8d52-63b874b0e9fd'
[2021-03-13T12:41:10.555Z] Reading host configuration file 'C:\Source\ESBPhase2\Test\AzureMock\figshare\host.json'
[2021-03-13T12:41:10.557Z] Host configuration file read:
[2021-03-13T12:41:10.558Z] {
[2021-03-13T12:41:10.560Z]   "version": "2.0",
[2021-03-13T12:41:10.598Z]   "logging": {
[2021-03-13T12:41:10.601Z]     "applicationInsights": {
[2021-03-13T12:41:10.669Z]       "samplingSettings": {
[2021-03-13T12:41:10.670Z]         "isEnabled": true,
[2021-03-13T12:41:10.749Z]         "excludedTypes": "Request"
[2021-03-13T12:41:10.751Z]       }
[2021-03-13T12:41:10.815Z]     }
[2021-03-13T12:41:10.844Z]   },
[2021-03-13T12:41:10.867Z]   "extensionBundle": {
[2021-03-13T12:41:10.874Z]     "id": "Microsoft.Azure.Functions.ExtensionBundle",
[2021-03-13T12:41:10.902Z]     "version": "[1.*, 2.0.0)"
[2021-03-13T12:41:10.919Z]   }
[2021-03-13T12:41:10.920Z] }
[2021-03-13T12:41:10.942Z] Loading functions metadata
[2021-03-13T12:41:10.964Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:java
[2021-03-13T12:41:10.967Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:node
[2021-03-13T12:41:10.988Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:python
[2021-03-13T12:41:10.994Z] Reading functions metadata
[2021-03-13T12:41:11.023Z] 1 functions found
[2021-03-13T12:41:11.054Z] 1 functions loaded
[2021-03-13T12:41:11.061Z] Looking for extension bundle Microsoft.Azure.Functions.ExtensionBundle at C:\Users\andrew\.azure-functions-core-tools\Functions\ExtensionBundles\Microsoft.Azure.Functions.ExtensionBundle
[2021-03-13T12:41:11.068Z] Found a matching extension bundle at C:\Users\andrew\.azure-functions-core-tools\Functions\ExtensionBundles\Microsoft.Azure.Functions.ExtensionBundle\1.5.1
[2021-03-13T12:41:11.088Z] Fetching information on versions of extension bundle Microsoft.Azure.Functions.ExtensionBundle available on https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/index.json
[2021-03-13T12:41:11.232Z] Skipping bundle download since it already exists at path C:\Users\andrew\.azure-functions-core-tools\Functions\ExtensionBundles\Microsoft.Azure.Functions.ExtensionBundle\1.5.1
[2021-03-13T12:41:11.239Z] Loading extension bundle from C:\Users\andrew\.azure-functions-core-tools\Functions\ExtensionBundles\Microsoft.Azure.Functions.ExtensionBundle\1.5.1\bin
[2021-03-13T12:41:11.243Z] Reading host configuration file 'C:\Source\ESBPhase2\Test\AzureMock\figshare\host.json'
[2021-03-13T12:41:11.247Z] Host configuration file read:
[2021-03-13T12:41:11.253Z] {
[2021-03-13T12:41:11.256Z]   "version": "2.0",
[2021-03-13T12:41:11.290Z]   "logging": {
[2021-03-13T12:41:11.294Z]     "applicationInsights": {
[2021-03-13T12:41:11.306Z]       "samplingSettings": {
[2021-03-13T12:41:11.308Z]         "isEnabled": true,
[2021-03-13T12:41:11.326Z]         "excludedTypes": "Request"
[2021-03-13T12:41:11.335Z]       }
[2021-03-13T12:41:11.379Z]     }
[2021-03-13T12:41:11.405Z]   },
[2021-03-13T12:41:11.430Z]   "extensionBundle": {
[2021-03-13T12:41:11.431Z]     "id": "Microsoft.Azure.Functions.ExtensionBundle",
[2021-03-13T12:41:11.449Z]     "version": "[1.*, 2.0.0)"
[2021-03-13T12:41:11.450Z]   }
[2021-03-13T12:41:11.451Z] }
[2021-03-13T12:41:11.659Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:java
[2021-03-13T12:41:11.662Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:node
[2021-03-13T12:41:11.664Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:python
[2021-03-13T12:41:11.744Z] Initializing Warmup Extension.
[2021-03-13T12:41:11.782Z] Initializing Host. OperationId: '16584ae7-9853-4a63-8d52-63b874b0e9fd'.
[2021-03-13T12:41:11.792Z] Host initialization: ConsecutiveErrors=0, StartupCount=1, OperationId=16584ae7-9853-4a63-8d52-63b874b0e9fd
[2021-03-13T12:41:11.816Z] LoggerFilterOptions
[2021-03-13T12:41:11.818Z] {
[2021-03-13T12:41:11.822Z]   "MinLevel": "None",
[2021-03-13T12:41:11.824Z]   "Rules": [
[2021-03-13T12:41:11.825Z]     {
[2021-03-13T12:41:11.826Z]       "ProviderName": null,
[2021-03-13T12:41:11.851Z]       "CategoryName": null,
[2021-03-13T12:41:11.853Z]       "LogLevel": null,
[2021-03-13T12:41:11.854Z]       "Filter": "<AddFilter>b__0"
[2021-03-13T12:41:11.855Z]     },
[2021-03-13T12:41:11.856Z]     {
[2021-03-13T12:41:11.856Z]       "ProviderName": "Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.SystemLoggerProvider",
[2021-03-13T12:41:11.857Z]       "CategoryName": null,
[2021-03-13T12:41:11.882Z]       "LogLevel": "None",
[2021-03-13T12:41:11.883Z]       "Filter": null
[2021-03-13T12:41:11.885Z]     },
[2021-03-13T12:41:11.886Z]     {
[2021-03-13T12:41:11.887Z]       "ProviderName": "Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.SystemLoggerProvider",
[2021-03-13T12:41:11.888Z]       "CategoryName": null,
[2021-03-13T12:41:11.901Z]       "LogLevel": null,
[2021-03-13T12:41:11.902Z]       "Filter": "<AddFilter>b__0"
[2021-03-13T12:41:11.915Z]     },
[2021-03-13T12:41:11.916Z]     {
[2021-03-13T12:41:11.917Z]       "ProviderName": "Azure.Functions.Cli.Diagnostics.ColoredConsoleLoggerProvider",
[2021-03-13T12:41:11.918Z]       "CategoryName": null,
[2021-03-13T12:41:11.919Z]       "LogLevel": null,
[2021-03-13T12:41:11.945Z]       "Filter": "<AddFilter>b__0"
[2021-03-13T12:41:11.947Z]     }
[2021-03-13T12:41:11.948Z]   ]
[2021-03-13T12:41:11.949Z] }
[2021-03-13T12:41:11.950Z] LoggerFilterOptions
[2021-03-13T12:41:11.951Z] {
[2021-03-13T12:41:11.977Z]   "MinLevel": "None",
[2021-03-13T12:41:11.979Z]   "Rules": [
[2021-03-13T12:41:11.980Z]     {
[2021-03-13T12:41:11.981Z]       "ProviderName": null,
[2021-03-13T12:41:11.982Z]       "CategoryName": null,
[2021-03-13T12:41:11.983Z]       "LogLevel": null,
[2021-03-13T12:41:11.983Z]       "Filter": "<AddFilter>b__0"
[2021-03-13T12:41:11.984Z]     },
[2021-03-13T12:41:11.985Z]     {
[2021-03-13T12:41:11.986Z]       "ProviderName": "Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.SystemLoggerProvider",
[2021-03-13T12:41:11.987Z]       "CategoryName": null,
[2021-03-13T12:41:11.988Z]       "LogLevel": "None",
[2021-03-13T12:41:11.988Z]       "Filter": null
[2021-03-13T12:41:11.989Z]     },
[2021-03-13T12:41:11.990Z]     {
[2021-03-13T12:41:11.991Z]       "ProviderName": "Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.SystemLoggerProvider",
[2021-03-13T12:41:11.992Z]       "CategoryName": null,
[2021-03-13T12:41:11.993Z]       "LogLevel": null,
[2021-03-13T12:41:11.994Z]       "Filter": "<AddFilter>b__0"
[2021-03-13T12:41:11.995Z]     },
[2021-03-13T12:41:11.995Z]     {
[2021-03-13T12:41:11.996Z]       "ProviderName": "Azure.Functions.Cli.Diagnostics.ColoredConsoleLoggerProvider",
[2021-03-13T12:41:11.997Z]       "CategoryName": null,
[2021-03-13T12:41:11.998Z]       "LogLevel": null,
[2021-03-13T12:41:12.023Z]       "Filter": "<AddFilter>b__0"
[2021-03-13T12:41:12.025Z]     }
[2021-03-13T12:41:12.026Z]   ]
[2021-03-13T12:41:12.058Z] }
[2021-03-13T12:41:12.073Z] FunctionResultAggregatorOptions
[2021-03-13T12:41:12.075Z] {
[2021-03-13T12:41:12.076Z]   "BatchSize": 1000,
[2021-03-13T12:41:12.078Z]   "FlushTimeout": "00:00:30",
[2021-03-13T12:41:12.080Z]   "IsEnabled": true
[2021-03-13T12:41:12.081Z] }
[2021-03-13T12:41:12.082Z] SingletonOptions
[2021-03-13T12:41:12.083Z] {
[2021-03-13T12:41:12.084Z]   "LockPeriod": "00:00:15",
[2021-03-13T12:41:12.085Z]   "ListenerLockPeriod": "00:00:15",
[2021-03-13T12:41:12.086Z]   "LockAcquisitionTimeout": "10675199.02:48:05.4775807",
[2021-03-13T12:41:12.088Z]   "LockAcquisitionPollingInterval": "00:00:05",
[2021-03-13T12:41:12.089Z]   "ListenerLockRecoveryPollingInterval": "00:01:00"
[2021-03-13T12:41:12.091Z] }
[2021-03-13T12:41:12.093Z] HttpOptions
[2021-03-13T12:41:12.117Z] {
[2021-03-13T12:41:12.119Z]   "DynamicThrottlesEnabled": false,
[2021-03-13T12:41:12.120Z]   "MaxConcurrentRequests": -1,
[2021-03-13T12:41:12.147Z]   "MaxOutstandingRequests": -1,
[2021-03-13T12:41:12.148Z]   "RoutePrefix": "api"
[2021-03-13T12:41:12.149Z] }
[2021-03-13T12:41:12.183Z] Starting JobHost
[2021-03-13T12:41:12.186Z] Starting Host (HostId=hepatica-952862876, InstanceId=b377e10e-37b4-4033-bd8f-bc111fe8478e, Version=3.0.15371.0, ProcessId=13528, AppDomainId=1, InDebugMode=False, InDiagnosticMode=False, FunctionsExtensionVersion=(null))
[2021-03-13T12:41:12.267Z] Loading functions metadata
[2021-03-13T12:41:12.270Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:java
[2021-03-13T12:41:12.275Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:node
[2021-03-13T12:41:12.277Z] FUNCTIONS_WORKER_RUNTIME set to powershell. Skipping WorkerConfig for language:python
[2021-03-13T12:41:12.284Z] Loading proxies metadata
[2021-03-13T12:41:12.306Z] Initializing Azure Function proxies
[2021-03-13T12:41:12.480Z] 0 proxies loaded
[2021-03-13T12:41:12.492Z] 1 functions loaded
[2021-03-13T12:41:12.516Z] Generating 1 job function(s)
[2021-03-13T12:41:12.534Z] Found the following functions:
[2021-03-13T12:41:12.536Z] Host.Functions.s
[2021-03-13T12:41:12.591Z]
[2021-03-13T12:41:12.617Z] Initializing function HTTP routes
[2021-03-13T12:41:12.635Z] Mapped function route 'api/s' [post] to 's'
[2021-03-13T12:41:12.636Z]
[2021-03-13T12:41:12.648Z] Host initialized (387ms)
[2021-03-13T12:41:12.667Z] Host started (416ms)
[2021-03-13T12:41:12.669Z] Job host started

Functions:

        s: [POST] http://localhost:7071/api/s

[2021-03-13T12:41:13.563Z] Worker process started and initialized.
```

and then when using the html at the top of the post to post to the Azure Function I get this response:

```
[2021-03-13T12:41:17.684Z] Host lock lease acquired by instance ID '00000000000000000000000057C78358'.
[2021-03-13T12:41:53.559Z] Executing HTTP request: {
[2021-03-13T12:41:53.562Z]   requestId: "950422d7-1048-4901-a72a-f80537f93ddc",
[2021-03-13T12:41:53.563Z]   method: "POST",
[2021-03-13T12:41:53.564Z]   userAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36 Edg/89.0.774.45",
[2021-03-13T12:41:53.565Z]   uri: "/api/s"
[2021-03-13T12:41:53.567Z] }
[2021-03-13T12:41:53.702Z] Executing 'Functions.s' (Reason='This function was programmatically called via the host APIs.', Id=8509714e-2875-4f32-85fd-5d38b2cdda05)
[2021-03-13T12:41:56.399Z] Executed 'Functions.s' (Succeeded, Id=8509714e-2875-4f32-85fd-5d38b2cdda05, Duration=2720ms)
[2021-03-13T12:41:56.413Z] Executed HTTP request: {
[2021-03-13T12:41:56.417Z]   requestId: "950422d7-1048-4901-a72a-f80537f93ddc",
[2021-03-13T12:41:56.418Z]   identities: "(WebJobsAuthLevel:Admin)",
[2021-03-13T12:41:56.460Z]   status: "400",
[2021-03-13T12:41:56.513Z]   duration: "2852"
[2021-03-13T12:41:56.589Z] }
```

Can anyone point me in the direction of where I'm going wrong or how I can get more logging to help find the cause?

Tried using PowerShell 7.1.2 and 5.1.0
Pode version 2.1.1

Thank you for any assistance
Andrew

**Comment by Badgerati** [comment]
Hey @andrewrdavidson,

So `multipart/form-data` should technically work however, unless you're trying upload files, you don't need to supply the `enctype` on `<form>` - which will default to `application/x-www-form-urlencoded`.

**Comment by andrewrdavidson** [comment]
Thanks @Badgerati. I'm using Pode to mock a specific service API which uses the multipart/form-data post method, so I really need to use that. I'll try without the enctype and see what happens. Is there anything else I can do to shed any light on this?

**Comment by Badgerati** [comment]
I've a vague memory of PowerShell Azure Functions always having the RawBody as a String, which would cause the multipart parser to fail. To check, https://github.com/Badgerati/Pode/blob/6fbb60c5302b408516a50a261bcee1f910a6462b/src/Private/Helpers.ps1#L1300, if the type of that is a String then that's the issue.

I'm currently reworking the multipart parser in #696, so I'll see if there's anything I can do there as well.

**Comment by Badgerati** [comment]
Got it, it was a string, but I've managed to fix it as a part of #696 😃

**Comment by andrewrdavidson** [comment]
Wonderful, thank you @Badgerati I will look forward to implementing this! :-)
